src/compiler/marks.ts

```diff
@@ -506,8 +512,10 @@ export function text(model: Model, layout, style) {
   // color should be set to background
   p.fill = {value: fieldDef.color};
 
-  var opacity = model.fieldDef(COLOR).opacity  || style.opacity;
-  if(opacity) p.opacity = {value: opacity};
+  var opacity = model.fieldDef(COLOR).opacity;
```

Comment:
Can you add a note here that this is where we decide to set or not to set the opacity?